### ISSUE TITLE
Warning: array_combine(): Both parameters should have an equal number of elements

### ISSUE THREAD

**Issue by hostep** [clerkio/clerk-magento2#27]
Hi!

This is using version 3.3.1 of this module on Magento OS 2.3.3

We are getting the following error when going to a search result page, for example: `/catalogsearch/result/?q=test`:

```
Exception #0 (Exception): Warning: array_combine(): Both parameters should have an equal number of elements in vendor/clerk/magento2/Block/Result.php on line 72
<pre>#1 array_combine() called at [vendor/clerk/magento2/Block/Result.php:72]
#2 Clerk\Clerk\Block\Result->getSpanAttributes() called at [generated/code/Clerk/Clerk/Block/Result/Interceptor.php:50]
#3 Clerk\Clerk\Block\Result\Interceptor->getSpanAttributes() called at [vendor/clerk/magento2/view/frontend/templates/result.phtml:4]
#4 include() called at [vendor/magento/framework/View/TemplateEngine/Php.php:59]
#5 Magento\Framework\View\TemplateEngine\Php->render() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#6 Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
...
```

If I output the `$titles` variable, I get:
```
Array
(
    [1] => Array
        (
            [sort_order] => 0
        )

    [0] => Array
        (
            [label] => 
        )

)
```

Calling `array_keys` on that variable results in 2 elements:
```
Array
(
    [0] => 1
    [1] => 0
)
```

Calling `array_column($titles, 'label')` on that variable results in 1 element:
```
Array
(
    [0] => 
)
```

So it feels like this code isn't robust enough.
Is this enough info for you guys to figure this out?

Thanks!

**Comment by duckchip** [comment]
I was able to reproduce this issue. 
Note that this only happens when no column is defined in the `clerk/faceted_search/titles` config value. 
If 1 or more columns are defined, it seem to work

**Comment by dpotkoc** [comment]
This issue is still existing in module version: 4.1.4

**Comment by CasperKN** [comment]
This should now be fixed in version 4.1.8, thanks for your feedback, we really appreciate it.
